### ISSUE TITLE
cargo tests failed when build rust 1.31

### ISSUE THREAD

**Issue by jeizsm** [rust-lang/cargo#6479]
**Problem**
Two tests fail on alpine linux when build rust 1.31
```
failures:

---- package::exclude stdout ----
running `/home/build/rust/src/rustc-1.31.1-src/build/x86_64-alpine-linux-musl/stage2-tools/x86_64-alpine-linux-musl/release/cargo package --no-verify -v`
thread 'package::exclude' panicked at '
Expected: execs
    but: differences:
 14 - |[WARNING] No (git) Cargo.toml found at `[..]` in workdir `[..]`|
    + |warning: No (git) VCS found for `/home/build/rust/src/rustc-1.31.1-src/build/x86_64-alpine-linux-musl/stage2-tools/x86_64-alpine-linux-musl/cit/t848/foo`|


other output:
``', tools/cargo/tests/testsuite/support/mod.rs:741:13
note: Some details are omitted, run with `RUST_BACKTRACE=full` for a verbose backtrace.
stack backtrace:
   0: std::sys::unix::backtrace::tracing::imp::unwind_backtrace
             at libstd/sys/unix/backtrace/tracing/gcc_s.rs:49
   1: std::sys_common::backtrace::print
             at libstd/sys_common/backtrace.rs:71
             at libstd/sys_common/backtrace.rs:59
   2: std::panicking::default_hook::{{closure}}
             at libstd/panicking.rs:211
   3: std::panicking::default_hook
             at libstd/panicking.rs:221
   4: std::panicking::rust_panic_with_hook
             at libstd/panicking.rs:476
   5: std::panicking::continue_panic_fmt
             at libstd/panicking.rs:390
   6: std::panicking::begin_panic_fmt
             at libstd/panicking.rs:345
   7: testsuite::support::Execs::run
   8: core::ops::function::FnOnce::call_once
   9: <F as alloc::boxed::FnBox<A>>::call_box
             at libtest/lib.rs:1468
             at /home/build/rust/src/rustc-1.31.1-src/src/libcore/ops/function.rs:238
             at /home/build/rust/src/rustc-1.31.1-src/src/liballoc/boxed.rs:672
  10: __rust_maybe_catch_panic
             at libpanic_unwind/lib.rs:102

---- package::include stdout ----
running `/home/build/rust/src/rustc-1.31.1-src/build/x86_64-alpine-linux-musl/stage2-tools/x86_64-alpine-linux-musl/release/cargo package --no-verify -v`
thread 'package::include' panicked at '
Expected: execs
    but: differences:
  2 - |[WARNING] No (git) Cargo.toml found at `[..]` in workdir `[..]`|
    + |warning: No (git) VCS found for `/home/build/rust/src/rustc-1.31.1-src/build/x86_64-alpine-linux-musl/stage2-tools/x86_64-alpine-linux-musl/cit/t852/foo`|


other output:
``', tools/cargo/tests/testsuite/support/mod.rs:741:13
note: Some details are omitted, run with `RUST_BACKTRACE=full` for a verbose backtrace.
stack backtrace:
   0: std::sys::unix::backtrace::tracing::imp::unwind_backtrace
             at libstd/sys/unix/backtrace/tracing/gcc_s.rs:49
   1: std::sys_common::backtrace::print
             at libstd/sys_common/backtrace.rs:71
             at libstd/sys_common/backtrace.rs:59
   2: std::panicking::default_hook::{{closure}}
             at libstd/panicking.rs:211
   3: std::panicking::default_hook
             at libstd/panicking.rs:221
   4: std::panicking::rust_panic_with_hook
             at libstd/panicking.rs:476
   5: std::panicking::continue_panic_fmt
             at libstd/panicking.rs:390
   6: std::panicking::begin_panic_fmt
             at libstd/panicking.rs:345
   7: testsuite::support::Execs::run
   8: core::ops::function::FnOnce::call_once
   9: <F as alloc::boxed::FnBox<A>>::call_box
             at libtest/lib.rs:1468
             at /home/build/rust/src/rustc-1.31.1-src/src/libcore/ops/function.rs:238
             at /home/build/rust/src/rustc-1.31.1-src/src/liballoc/boxed.rs:672
  10: __rust_maybe_catch_panic
             at libpanic_unwind/lib.rs:102
```

**Steps**
1. Compile rust 1.31 on alpine linux with rust 1.30

**Notes**

Alpine linux edge, rust 1.30, rust 1.31

**Comment by ehuss** [comment]
Thanks for the report! This has been fixed in #6260.  1.32 should not have this problem.  I think one workaround is to build inside a git repo. Or you can apply the patch in that PR.

**Comment by jeizsm** [comment]
Thank you
